### ISSUE TITLE
[a11y]: activity accessibility violations in data table component

### ISSUE THREAD

**Issue by jingx-xue** [carbon-design-system/carbon#11582]
### Package

carbon-components, carbon-components-react

### Browser

Chrome

### Operating System

Windows

### Package version

10.43.0

### React version

16.8.6

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

![image](https://user-images.githubusercontent.com/48215968/172677007-c1d36b4e-8b5d-437b-9b6e-23a9db43e3a8.png)
![image](https://user-images.githubusercontent.com/48215968/172677057-3e7ea26b-faf0-4b81-bc75-b9706bff8432.png)

data table with expand function showing the warning of 1.3.1 Info and Relationships

### WCAG 2.1 Violation

1.3.1 Info and Relationships

### Reproduction/example

https://stackblitz.com/edit/github-exn3kb?file=src%2FApp.jsx

### Steps to reproduce

Open Sandbox example
Run accessibility tool

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by joshblack** [comment]
Hey @jingx-xue! This may be due to an issue in the example related to the usage of a `div`. Here's an updated example: https://stackblitz.com/edit/github-exn3kb-zlc8fs?file=src%2FApp.jsx 

Let me know if that works for you!

**Comment by joshblack** [comment]
Hey there! Going to close this out due to inactivity. Feel free to comment and we can re-open this issue 👍
